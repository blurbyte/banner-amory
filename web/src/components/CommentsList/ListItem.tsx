import * as React from 'react';
import styled from 'styled-components';

const Wrapper = styled.li`
  display: grid;
  grid-template-columns: 15rem 1fr;
  height: 3rem;
  align-items: end;
`;

const Name = styled.span`
  font-weight: 600;
  padding-right: 1rem;
`;

const Message = styled.span`
  font-weight: 400;
`;

type ListItemProps = {
  userName: string;
  message: string;
};

class ListItem extends React.Component<ListItemProps> {
  render() {
    const { userName, message } = this.props;

    return (
      <Wrapper>
        <Name>{userName}</Name>
        <Message>{message}</Message>
      </Wrapper>
    );
  }
}

export default ListItem;
