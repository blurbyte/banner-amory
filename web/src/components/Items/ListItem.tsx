// single item in the grid

import * as React from 'react';
import { Link as BaseLink } from '@reach/router';
import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '../../styles/colors';
import { ItemBasic as ListItemProps } from '../../types/Item';
import Image from '../Image';

const Item = styled.li`
  position: relative;
  background: ${transparentize(0.9, Colors.blue)};
  z-index: 99;
  user-select: none;
`;

const Link = styled(BaseLink)`
  color: ${Colors.blue};
`;

const Rank = styled.p`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: ${Colors.white};
  margin: 0;
  padding: 0;
  width: 2rem;
  height: 2rem;
  font-weight: 600;
`;

class ListItem extends React.Component<ListItemProps> {
  render() {
    const { slug, rank } = this.props;

    return (
      <Item>
        <Link to={`item/${slug}`}>
          <Image slug={slug} />
          <Rank>{rank}</Rank>
        </Link>
      </Item>
    );
  }
}

export default ListItem;
