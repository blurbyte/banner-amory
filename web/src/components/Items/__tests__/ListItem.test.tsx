import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import ListItem from '../ListItem';

test('renders correctly', () => {
  const props = {
    slug: 'test-slug',
    rank: 10
  };
  const component = renderer.create(<ListItem {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
