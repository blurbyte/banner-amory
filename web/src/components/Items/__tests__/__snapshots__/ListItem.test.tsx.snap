// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  max-width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  position: relative;
  background: rgba(61,38,218,0.1);
  z-index: 99;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  color: #3d26da;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: #ffffff;
  margin: 0;
  padding: 0;
  width: 2rem;
  height: 2rem;
  font-weight: 600;
}

<li
  className="c0"
>
  <a
    className="c1"
    href="/item/test-slug"
    onClick={[Function]}
  >
    <img
      className="c2"
      src="undefinedtest-slug.png"
    />
    <p
      className="c3"
    >
      10
    </p>
  </a>
</li>
`;
